### PR TITLE
Build wasm package

### PR DESCRIPTION
**Pull request by dfrankland**
depends on #135

### CODE REVIEW

File: context-js/src/lib.rs [hunk 1]
```diff
@@ -37,8 +37,8 @@ pub fn env_parse(env_vars: js_sys::Object) -> Result<Option<env::parser::CIInfo>
 }
 
 #[wasm_bindgen]
-pub fn env_validate(ci_info: env::parser::CIInfo) -> env::validator::EnvValidation {
-    env::validator::validate(&ci_info)
+pub fn env_validate(ci_info: &env::parser::CIInfo) -> env::validator::EnvValidation {
+    env::validator::validate(ci_info)
 }
```
Comment (dfrankland) [inline context-js/src/lib.rs:42]: Turns out that only references work as arguments for wasm-exported functions

File: context/src/repo/mod.rs [hunk 2]
```diff
@@ -186,6 +186,37 @@ impl BundleRepo {
     }
 }
 
+#[cfg(feature = "wasm")]
+#[wasm_bindgen]
+impl BundleRepo {
+    #[wasm_bindgen(constructor)]
+    pub fn js_new(
```
Comment (dfrankland) [inline context/src/repo/mod.rs:193]: These constructors are needed to use `new` in JS

File: .github/workflows/pull_request.yml [hunk 3]
```diff
@@ -103,6 +103,9 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
+      - name: Setup and build wasm
+        uses: ./.github/actions/setup_build_wasm
+
```
Comment (TylerJang27) [inline .github/workflows/pull_request.yml:108]: Sanity check: is the intention here to build the wasm targets and then run eslint on them in CI? Is there a specific pattern/issue that this is designed to catch (other than just making sure they build as part of a PR workflow)?

File: context-js/tests/parse_and_validate.test.ts
Comment (TylerJang27) [inline context-js/tests/parse_and_validate.test.ts:1]: 🏆

**Review by TylerJang27 — APPROVED** [review]
💯

File: .github/workflows/pull_request.yml (hunk 3 above)
Comment (dfrankland) [reply .github/workflows/pull_request.yml:108]: I'm running lints that take into account types on the test code. The test code relies on types from the wasm build, so I have to build the wasm package prior to linting.

**Pull request merged**